checkov/terraform/runner.py

```diff
@@ -116,47 +117,73 @@ def run(
         parsing_errors: dict[str, Exception] = {}
         self.load_external_checks(external_checks_dir)
         local_graph = None
-
+        all_graphs = []
         if self.context is None or self.definitions is None or self.breadcrumbs is None:
             self.definitions = {}
             logging.info("Scanning root folder and producing fresh tf_definitions and context")
             if root_folder:
                 root_folder = os.path.abspath(root_folder)
-
-                local_graph, self.definitions = self.graph_manager.build_graph_from_source_directory(
-                    source_dir=root_folder,
-                    local_graph_class=self.graph_class,
-                    download_external_modules=runner_filter.download_external_modules,
-                    external_modules_download_path=runner_filter.external_modules_download_path,
-                    parsing_errors=parsing_errors,
-                    excluded_paths=runner_filter.excluded_paths,
-                    vars_files=runner_filter.var_files,
-                    create_graph=CHECKOV_CREATE_GRAPH,
-                )
+                tf_split_graph = strtobool(os.getenv('TF_SPLIT_GRAPH', 'False'))
+                if tf_split_graph:
+                    graphs_with_definitions = self.graph_manager.build_multi_graph_from_source_directory(
+                        source_dir=root_folder,
+                        local_graph_class=self.graph_class,
+                        download_external_modules=runner_filter.download_external_modules,
+                        external_modules_download_path=runner_filter.external_modules_download_path,
+                        parsing_errors=parsing_errors,
+                        excluded_paths=runner_filter.excluded_paths,
+                        vars_files=runner_filter.var_files,
+                        create_graph=CHECKOV_CREATE_GRAPH,
+                    )
+                    local_graph = []
+                    for graph, definitions in graphs_with_definitions:
+                        for definition in definitions:
+                            self.definitions.update(definition)
```

Comment:
so just to make sure - basically we still have one runner with all of the definitions, but multiple graphs.

Reply:
Yes multiple graphs and definitions is still a dict as before.